src/apps/frontend/components/task/task-popup.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+
+interface TaskPopupProps {
+  popupOpen: boolean;
+  setPopupOpen: (open: boolean) => void;
+}
+
+const TaskPopup: React.FC<TaskPopupProps> = ({ popupOpen, setPopupOpen }) => {
```

Comment:
I think this is TaskModal, please see difference between modal and popup.

Also, is this copied from TailwinAdmin, there is things that should be improved in the React code

Reply:
I will correct this to TaskModal.